sigsci_site_manager/user.py

```diff
@@ -0,0 +1,166 @@
+                                                              member[cols[1]]['name'])))
+            print_sorted_array(line_entries)
+
+
+def do_add_user(args):
+    api = init_api(args.username, args.password, args.token, args.corp,
+                   args.dry_run)
+
+    input_file = resolve_input_file(args.file_name)
+    location_string = 'corp'
+    if args.site_name:
+        api.site = args.site_name
+        location_string = args.site_name
+
+    if args.email_id:
+        data = resolve_user_data_role(api, args.role_name)
+        if args.dry_run:
+            print("Adding %s to %s with %s role" %
+                  (args.email_id, location_string, data['role']))
+        add_new_user(api, args.email_id, args.role_name, args.api_user)
+    if input_file:
+        add_users_from_file(input_file, api, args, location_string)
+
+
+def add_users_from_file(input_file, api, args, location_string):
+    entries = 0
+    with open_file_for_read(input_file) as f:
+        line = f.readline()
+        while line:
+            cols = line.split(",")
+            if len(cols) == 2:
+                if args.role_name:
+                    role = args.role_name
+                else:
+                    role = cols[1]
+                data = resolve_user_data_role(api, role)
+                args.role_name = cols[1]
+                args.email_id = cols[0]
+
+                if args.dry_run:
+                    print("Adding %s to %s with %s role" %
+                          (cols[0].strip(), location_string, data['role']))
+                add_new_user(api, args.email_id,
+                             args.role_name, args.api_user)
+
+                entries += 1
+            line = f.readline()
+    print("Processed %d entries" % entries)
+
+
+def do_remove_user(args):
+    api = init_api(args.username, args.password, args.token, args.corp,
+                   args.dry_run)
+
+    input_file = resolve_input_file(args.file_name)
+    location_string = 'corp'
+    if args.site_name:
+        api.site = args.site_name
+        location_string = args.site_name
+
+    if args.email_id:
+        if args.dry_run:
+            print("Deleting %s from %s" % (args.email_id, location_string))
+        else:
+            if args.site_name:
+                api.delete_site_member(args.email_id)
+            else:
+                api.delete_corp_user(args.email_id)
+    if input_file:
+        entries = 0
+        with open_file_for_read(input_file) as f:
+            line = f.readline()
+            while line:
+                if args.dry_run:
+                    print("Deleting %s from %s" %
+                          (line.strip(), location_string))
+                else:
+                    if args.site_name:
+                        api.delete_site_member(line.strip())
+                    else:
+                        api.delete_corp_user(line.strip())
```

Comment:
Same as above. Instead of calling these delete methods in different places read the file to generate a list to delete and then only call the delete methods in one place.

Reply:
addressed in comit 340cc86